modules/exploits/linux/http/zyxel_ztp_rce.rb

```diff
@@ -0,0 +1,156 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Zyxel Firewall ZTP Unauthenticated Command Injection',
+        'Description' => %q{
+          This module exploits CVE-2022-30525, an unauthenticated and remote
+          command injection vulnerability affecting Zyxel firewalls with zero
+          touch provisioning (ZTP) support. This module exploits a command injection
+          exposed by the /ztp/cgi-bin/handler when handling setWanPortSt commands.
+          The module adds OS commands to the mtu field of the setWanPortSt request
+          in order to achieve command execution as `nobody`.
+
+          Affected Zyxel models are:
+
+          * USG FLEX 50, 50W, 100W, 200, 500, 700 using firmware 5.21 and below
+          * USG20-VPN and USG20W-VPN using firmware 5.21 and below
+          * ATP 100, 200, 500, 700, 800 using firmware 5.21 and below
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'jbaines-r7' # Vulnerability discovery and Metasploit module
+        ],
+        'References' => [
+          [ 'CVE', '2022-30525' ],
+        ],
+        'DisclosureDate' => '2022-04-28',
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_MIPS64,],
+        'Privileged' => true,
```

Comment:
Executed as `nobody`.

```suggestion
        'Privileged' => false,
```